components/script/dom/window.rs

```diff
@@ -137,10 +137,15 @@ use webrender_api::{DocumentId, ExternalScrollId, RenderApiSender};
 use webvr_traits::WebVRMsg;
 
 /// Current state of the window object
-#[derive(Clone, Copy, Debug, JSTraceable, MallocSizeOf, PartialEq)]
+#[derive(JSTraceable, MallocSizeOf)]
+#[must_root]
 enum WindowState {
-    Alive,
-    Zombie, // Pipeline is closed, but the window hasn't been GCed yet.
+    /// A window with a browsing-context.
+    Alive(WindowStateData),
```

Comment:
We could go for `BrowsingContextDiscarded` and `WithBrowsingContext`, because the switch really occurs as part of https://html.spec.whatwg.org/multipage/window-object.html#garbage-collection-and-browsing-contexts and the window and document could still be reachable.